osu.Game/Audio/PreviewTrack.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using osu.Framework.Audio.Track;
+
+namespace osu.Game.Audio
+{
+    public class PreviewTrack
+    {
+        public readonly Track Track;
+        private readonly Action<PreviewTrack> onStart;
```

Comment:
There shouldn't be two action delegates here. Just remove these and use the public ones?

Reply:
Don't understand which public delegates do you mean.

Reply:
Started and Stopped, just below.

Reply:
`onStart` is used for stopping current track before starting the new one. `Started` and `Stopped` events are fired after the new track was started.

Reply:
Is there a valid reason they can't be combined?

Reply:
I will try to combine them.